File: app/src/main/java/com/mithril/mobilegoldenleaf/database/dao/OrderDao.java
package com.mithril.mobilegoldenleaf.database.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.mithril.mobilegoldenleaf.models.Order;

import java.util.List;

@Dao
public interface OrderDao {
    @Insert
    long save(Order order);

    @Update
    void edit(Order order);

    @Query("SELECT * FROM `Order`")
    List<Order> all();

}

File: app/src/main/java/com/mithril/mobilegoldenleaf/persistence/repository/OrderDao.kt
package com.mithril.mobilegoldenleaf.persistence.repository

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

import com.mithril.mobilegoldenleaf.models.Order

@Dao
interface OrderDao {
    @Insert
    fun save(order: Order): Long

    @Update
    fun edit(order: Order)

    @Query("SELECT * FROM `Order`")
    fun all(): List<Order>

}
